bin/i18n/sync-in.rb

```diff
@@ -55,10 +55,11 @@ def get_i18n_strings(level)
 
     # authored_hints
     if level.authored_hints
+      authored_hints_string = 'authored_hints'
```

Comment:
this seems unnecessary, both here and below. I would _much_ prefer we use the hash keys directly

Reply:
Can you explain the preference? I agree the variable doesn't improve /too/ much, but it seems like using the same raw strings in different places introduces more risk of mis-alignment.

Reply:
Hiding the actual keys being used under a variable probably does reduce the risk of misalignment a bit in theory, but in practice there's no real difference between introducing a typo into `i18n_strings[authored_hints_string]` vs `i18n_strings['authored_hints']`; both will manifest as an error that will be easily noticeable. More importantly, though, it makes the code significantly more difficult to read. Interacting with hashes by passing strings and symbols as keys is standard; when I see a variable being used I expect that we're interacting with the hash programmatically rather than manually (as is the case here), and have to go searching for the variable declaration to realize that no, we're actually not doing anything programmatic here but are instead just making it look like we are.

If you did _really_ want to avoid using strings, then the right way to do it would be to declare these variables as constants in some more-globally-accessible spot in the code.

Reply:
Got it. I would like those to be globally accessible constants, but will count them as out of scope for this PR. Thanks for the explanation!